test/resource.cpp

```diff
@@ -51,6 +51,51 @@ TEST(CampResource, GetPlatform)
   ASSERT_EQ(static_cast<const Resource>(Omp()).get_platform(), Platform::omp_target);
 #endif
 }
+TEST(CampResource, Compare)
+{
+  Resource h1{Host()};
+  Resource h2{Host()};
+  Host h; Resource h3{h};
+
+#ifdef CAMP_HAVE_CUDA
+  Resource r1{Cuda()};
+  Resource r2{Cuda()};
+  Cuda s; Resource r3{s};
+#endif
+#ifdef CAMP_HAVE_HIP
+  Resource r1{Hip()};
+  Resource r2{Hip()};
+  Hip s; Resource r3{s};
+#endif
+#ifdef CAMP_HAVE_OMP_OFFLOAD
+  Resource r1{Omp()};
+  Resource r2{Omp()};
+  Omp s; Resource r3{s};
+#endif
+
+  ASSERT_TRUE(h1 == h1);
```

Comment:
Add tests comparing concrete resources like h and s.

Reply:
That would require an operator overload that could handle a `camp::resources::v1::Cuda != camp::resources::v1::Host` comparison and I currently don't have that functionality... is it possible to really compare a `T t` and `T t2`? That seems strange - or maybe i'm missing something

Reply:
Right, how about just comparing like with like?